### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by bramtayl**
Small changes to make sure everything works. I also added samplerate float conversion.

### CODE REVIEW

File: .github/workflows/Tests.yml [hunk 1]
```diff
@@ -18,8 +18,6 @@ jobs:
       matrix:
         version:
           - '1.3'
```
Comment (JeffFessler) [inline .github/workflows/Tests.yml:20]: Given that 1.6 is now the LTSR, how about testing 1.6, 1, nightly?  (and leave 1.3 behind?)

File: README.md [hunk 2]
```diff
@@ -75,27 +78,26 @@ end
...
 ```julia
-julia> using PortAudio, SampledSignals, LibSndFile
+julia> using PortAudio, SampledSignals, LibSndFile, FileIO
 
-julia> stream = PortAudioStream("Built-in Microph", 2, 0)
-PortAudio.PortAudioStream{Float32,SIUnits.SIQuantity{Int64,0,0,-1,0,0,0,0,0,0}}
-  Samplerate: 48000 s⁻¹
-  Buffer Size: 4096 frames
-  2 channel source: "Built-in Microph"
+julia> stream = PortAudioStream("default", 2, 0)
```
Comment (JeffFessler) [inline README.md:91]:
```suggestion
julia> stream = PortAudioStream("default", 1, 0)
```

FWIW, my iMac (despite its cost) has only a 1-channel microphone.
So I'd vote for a 1 channel example here to aid accessibility, but it's also OK as is...
(I kind of wish there were a "maximum number of channels" default, but another day.)

File: src/PortAudio.jl [hunk 3]
```diff
@@ -730,29 +730,35 @@ function PortAudioStream(
                     output_device.output_bounds.high_latency,
                 )
             end
-            if samplerate === nothing
-                samplerate = combine_default_sample_rates(
+            samplerate = if samplerate === nothing
```
Comment (JeffFessler) [inline src/PortAudio.jl:733]: Interesting, I'd never seen this construct before but now I see that it works.
I probably would have used `isnothing` and the ternary operator `?` but it seems fine.

**Comment by JeffFessler** [comment]
I made a few minor suggestions, but otherwise it LGTM.

File: .github/workflows/Tests.yml (hunk 1 above)
Comment (bramtayl) [reply .github/workflows/Tests.yml:20]: Sure, that would be a separate PR though. I messed up with git here; the first two commits are already on master. I'll try again on top of #102

File: README.md [hunk 4]
```diff
@@ -27,13 +27,16 @@ You can get a list of your system's devices with the `PortAudio.devices()` funct
...
+14-element Vector{PortAudio.PortAudioDevice}:
+ "sof-hda-dsp: - (hw:0,0)" 2→2
+ "sof-hda-dsp: - (hw:0,3)" 0→2
+ "sof-hda-dsp: - (hw:0,4)" 0→2
+ "sof-hda-dsp: - (hw:0,5)" 0→2
+ ⋮
+ "upmix" 8→8
+ "vdownmix" 6→6
+ "dmix" 0→2
+ "default" 32→32
```
[the remark this answers is not in the document]
Comment (bramtayl) [reply README.md:39]: And not many of them are that useful...

File: src/PortAudio.jl (hunk 3 above)
Comment (bramtayl) [reply src/PortAudio.jl:733]: Ok. I like doing it this way because, unlike the ternary operator, you can add multiple lines if you want.

**Comment by bramtayl** [comment]
Any better?

File: README.md [hunk 5]
```diff
@@ -10,10 +10,10 @@ PortAudio.jl is a wrapper for [libportaudio](http://www.portaudio.com/), which g
 
 ## Opening a stream
 
-The easiest way to open a source or sink is with the default `PortAudioStream()` constructor, which will open a 2-in, 2-out stream to your system's default device(s). The constructor can also take the input and output channel counts as positional arguments, or a variety of other keyword arguments.
+The easiest way to open a source or sink is with the default `PortAudioStream()` constructor, which will open a 2-in, 2-out stream to your system's default device(s). The constructor can also take the input and output channel counts as positional arguments, or a variety of other keyword arguments. If you don't specify latency and sample rate, PortAudio will use device defaults.
```
Comment (JeffFessler) [inline README.md:13]:
```suggestion
The easiest way to open a source or sink is with the default `PortAudioStream()` constructor, 
which will open a 2-in, 2-out stream to your system's default device(s).
The constructor can also take the input and output channel counts as positional arguments,
or a variety of other keyword arguments.
If named keyword arguments `latency` or `samplerate` are unspecified, then PortAudio will use device defaults.
```
(suggestion)

File: README.md [hunk 6]
```diff
@@ -10,10 +10,10 @@ PortAudio.jl is a wrapper for [libportaudio](http://www.portaudio.com/), which g
...
 
 ```julia
-PortAudioStream(inchans=2, outchans=2; eltype=Float32, samplerate=48000Hz, latency=0.1, synced=false)
+PortAudioStream(inchans=2, outchans=2; eltype=Float32, latency = nothing, samplerate = nothing)
```
Comment (JeffFessler) [inline README.md:16]:
```suggestion
PortAudioStream(inchans=2, outchans=2; eltype=Float32, samplerate=48000Hz, latency=0.1, synced=false)
PortAudioStream(inchans=2, outchans=2)
```

If the goal is to show the default behavior, then I don't think we need `eltype=Float32, latency = nothing, samplerate = nothing` do we?
If the goal is to show the options, then isn't it more informative to show the options with useful values like
`samplerate=48000Hz, latency=0.1` ?
Perhaps we need two examples here, one with defaults and one with informative values?
Perhaps I am missing why you wanted to change this example?

File: src/PortAudio.jl [hunk 7]
```diff
@@ -730,29 +730,35 @@ function PortAudioStream(
...
             if latency === nothing
                 latency = output_device.output_bounds.high_latency
             end
-            if samplerate === nothing
-                samplerate = output_device.default_sample_rate
+            samplerate = if samplerate === nothing
+                output_device.default_sample_rate
+            else
+                float(samplerate)
             end
```
Comment (JeffFessler) [inline src/PortAudio.jl:762]:
```suggestion
                    output_device.default_sample_rate
                else
                    float(samplerate)
                end
```
It seems unusual to me to have `else` and `end` in a column without a matching `if` in the same column.
But it's not that important so if you prefer your original way it's ok.

File: src/PortAudio.jl (hunk 3 above)
Comment (JeffFessler) [reply src/PortAudio.jl:733]: Just FYI it is possible to use multiple lines with ternary operators:
https://github.com/JeffFessler/MIRT.jl/blob/1be5a795a6ddb56f1478740a65033f9cf6caad55/src/fbp/sino_geom.jl#L543-L545

**Comment by JeffFessler** [comment]
> Any better?

oops sorry, i didn't realize you were waiting for feedback for me.
i've made a few more suggestions for your consideration.
i promise not to drag it out much longer after this :)

File: src/PortAudio.jl (hunk 7 above)
Comment (bramtayl) [reply src/PortAudio.jl:762]: Doesn't matter much to be but ok.

File: .github/workflows/Tests.yml (hunk 1 above)
Comment (JeffFessler) [reply .github/workflows/Tests.yml]: Agreed, that would be separate PR.  It was just a note for the future...

File: README.md (hunk 6 above)
Comment (JeffFessler) [reply README.md:20]: Oh gosh, I didn't realize that is why you were updating it.  I like your suggestion of removing Hz and synced, thereby showing the named keyword arguments.

File: README.md (hunk 2 above)
Comment (JeffFessler) [reply README.md]: Oh, I had overlooked the `maximum` option in the docs.  Thanks!

**Pull request merged**